homeassistant/loader.py

```diff
@@ -290,13 +290,22 @@ def resolve_from_root(
                 )
                 continue
 
-            return cls(
+            integration = cls(
                 hass,
                 f"{root_module.__name__}.{domain}",
                 manifest_path.parent,
                 manifest,
             )
 
+            if not integration.is_built_in:
+                try:
+                    validate_custom_integration_version(integration)
+                except IntegrationNotFound:
```

Comment:
When an this raise Integration not found ?

Reply:
If this is only used here, let's just make it return a boolean if valid

Reply:
I'll just drop the validate_custom_integration_version function, then it make sense

Reply:
And if only used here, maybe we should also prefix it with `_` to indicate it's a private API

Reply:
It did so little it did not really make sense to have it out of where it was used